src/mlpack/core/optimizers/adam/adam_update.hpp

```diff
@@ -0,0 +1,148 @@
+/**
+ * @file adam_update.hpp
+ * @author Ryan Curtin
+ * @author Vasanth Kalingeri
+ * @author Marcus Edel
+ * @author Vivek Pal
+ *
+ * Adam optimizer. Adam is an an algorithm for first-order gradient-based
+ * optimization of stochastic objective functions, based on adaptive estimates
+ * of lower-order moments.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_ADAM_ADAM_UPDATE_HPP
+#define MLPACK_CORE_OPTIMIZERS_ADAM_ADAM_UPDATE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * Adam is an optimizer that computes individual adaptive learning rates for
+ * different parameters from estimates of first and second moments of the
+ * gradients as given in the section 7 of the following paper.
+ *
+ * For more information, see the following.
+ *
+ * @code
+ * @article{Kingma2014,
+ *   author    = {Diederik P. Kingma and Jimmy Ba},
+ *   title     = {Adam: {A} Method for Stochastic Optimization},
+ *   journal   = {CoRR},
+ *   year      = {2014}
```

Comment:
@rcurtin  I'll get this covered but I'm not sure if we really want to add `url` here because [citation guidelines](https://github.com/mlpack/mlpack/wiki/DesignGuidelines#citations) suggest otherwise. Although, I'm aware that you wrote all design guidelines for mlpack but just wanted to check if there was a decision in the recent past to also add urls in the citations :)